app/views/layouts/_admin_dropdown.html.erb

```diff
@@ -1,178 +1,40 @@
-        <%= link_to(models_name(Category), admin_categories_path) %>
-      </li>
-    <% end %>
-
-
-    <% if can_administrate? :manage, Contact %>
-      <li class="dropdown-submenu">
-        <%= link_to(model_name(Contact), admin_contacts_path) %>
-      </li>
-    <% end %>
-
-    <% if can_administrate? :manage, Album %>
-      <li class="dropdown-submenu">
-        <%= link_to(model_name(Album), admin_gallery_albums_path)%>
-      </li>
-    <% end %>
-
-    <% if can_administrate? :manage, Page %>
-      <li class="dropdown-submenu">
-        <%= link_to(model_name(Page), admin_pages_path)%>
-      </li>
-    <% end %>
-
-    <% if can_administrate? :index, ShortLink %>
-      <li class="dropdown-submenu">
-        <%= link_to(models_name(ShortLink), short_links_path) %>
-      </li>
-    <% end %>
-
-    <% if can_administrate? :index, MailAlias %>
-      <li class="dropdown-submenu">
-        <%= link_to(models_name(MailAlias), mail_aliases_path) %>
-      </li>
-    <% end %>
-
-    <% if can_administrate? :index, Rent %>
-      <li class="dropdown-submenu">
-        <%= link_to(model_name(Rent), admin_rents_path)%>
-      </li>
-    <% end %>
-
-    <% if can_administrate? :index, Tool %>
-      <li class="dropdown-submenu">
-        <%= link_to(model_name(Tool), admin_tools_path)%>
-      </li>
-    <% end %>
-
-    <% if can_administrate? :index, Election %>
-      <li class="dropdown-submenu">
-        <%= link_to(model_name(Election), admin_elections_path)%>
-      </li>
-    <% end %>
-
-    <% if can_administrate? :index, Event %>
-      <li class="dropdown-submenu">
-        <%= link_to(model_name(Event), admin_events_path) %>
-      </li>
-    <% end %>
-
-    <% if can_administrate? :index, News %>
-      <li class="dropdown-submenu">
-        <%= link_to(model_name(News), admin_news_index_path) %>
-      </li>
-    <% end %>
-
-    <% if can_administrate? :index, Notice %>
-      <li class="dropdown-submenu">
-        <%= link_to(models_name(Notice), admin_notices_path)%>
-      </li>
-    <% end %>
-
-    <% if can_administrate? :index, Document %>
-      <li class="dropdown-submenu">
-        <%= link_to(model_name(Document), admin_documents_path) %>
-      </li>
-    <% end %>
-
-    <% if can_administrate? :index, Council %>
-      <li class="dropdown-submenu">
-        <%= link_to(model_name(Council), admin_councils_path) %>
-      </li>
-    <% end %>
-
-    <% if can_administrate? :manage, Council %>
-      <li class="dropdown-submenu">
-        <%= link_to(models_name(Post), '#') %>
-        <ul class="dropdown-menu">
-          <% Council.by_title.each do |council| %>
-            <% if can_administrate? :manage, council %>
-              <li><%= link_to(council, admin_council_posts_path(council)) %></li>
+  <ul class="dropdown-menu megamenu row">
+  <% all_privilegies.each do |c| %>
+    <li class="admin-menu-category">
+      <ul class="">
+        <% privilegies = check_privilegies(c) %>
```

Comment:
This would become `privileges = check_privileges(items)`. No need to send in and work with the title/header name!